test/TestServer.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const P = require('bluebird');
+const ServiceRunner = require('service-runner');
+
+class TestServer {
+    constructor(configPath) {
+        if (!configPath) {
+            throw new Error('Config path must be provided to the test runner');
+        }
+        this._configPath = configPath;
+        this._running = false;
+        this._runner = new ServiceRunner({
+            configFile: this._configPath
+        });
+        this._startupRetriesRemaining = 3;
+    }
+
+    start() {
+        if (this._running) {
+            console.log('The test server is already running. Skipping start.');
+            return P.resolve();
+        }
+
+        return this._runner.start()
+        .tap(() => {
+            this._running = true;
```

Comment:
Shouldn't this flag be set only if we actually managed to start the server? If the start fails, `.catch()` is activated, which calls the method again, but since the flag is set, it won't actually do anything, would it?

Reply:
`.tap` is just like `.then` so if it fails this will not be called.